anvill/src/Providers/TypeProvider.cpp

```diff
@@ -70,6 +73,28 @@ ProgramTypeProvider::TryGetFunctionType(uint64_t address) {
   return *decl;
 }
 
+std::optional<GlobalVarDecl>
+ProgramTypeProvider::TryGetVariableType(uint64_t address,
+                                        const llvm::DataLayout &layout) {
+  if (auto var_decl = program.FindVariable(address); var_decl) {
+
+    // Check integrity of the var_decl
+    CHECK_NOTNULL(var_decl->type);
+    CHECK_EQ(var_decl->address, address);
+    return *var_decl;
+
+  // if FindVariable fails to get the variable at address; get the variable
+  // containing the address
+  } else if (auto var_decl = program.FindInVariable(address, layout);
+             var_decl) {
+    CHECK_NOTNULL(var_decl->type);
+    CHECK_EQ(var_decl->address, address);
```

Comment:
`CHECK_LE`?

Reply:
I removed the checks here. It may not be necessary.